src/context.cc

```diff
@@ -636,6 +614,40 @@ WasmResult ContextBase::setTimerPeriod(std::chrono::milliseconds period,
   return WasmResult::Ok;
 }
 
+FilterHeadersStatus ContextBase::convertVmCallResultToFilterHeadersStatus(uint64_t result) {
+  if (stop_iteration_) {
+    stop_iteration_ = false;
+    return FilterHeadersStatus::StopIteration;
+  } else if (result > static_cast<uint64_t>(FilterHeadersStatus::StopAllIterationAndWatermark)) {
+    return FilterHeadersStatus::StopAllIterationAndWatermark;
```

Comment:
Any reason for not pulling `stop_iteration_` case here? Does it break if we return `StopAllIterationAndWatermark`?

Reply:
I just followed the implementation of router https://github.com/envoyproxy/envoy/blob/master/source/common/router/router.cc#L367-L369 but I verified it's working even if return `StopAllIterationAndWatermark`.